core-backend/sandbox/src/env.rs

```diff
@@ -154,7 +135,24 @@ impl<E: Ext + Into<ExtInfo> + 'static> Environment<E> for SandboxEnvironment<E>
             }
         })?;
 
-        let res = instance.invoke(entry_point, &[], &mut runtime);
+        self.runtime.replace(runtime);
+        self.instance.replace(instance);
+
+        Ok(())
+    }
+
+    /// Setup external environment and run closure.
+    ///
+    /// Setup external environment by providing `ext`, run nenwly initialized instance created from
+    /// provided `module`, do anything inside a `func` delegate.
+    ///
+    /// This will also set the beginning of the memory region to the `static_area` content _after_
+    /// creatig instance.
+    fn execute(&mut self, entry_point: &str) -> Result<BackendReport, BackendError> {
+        let instance = self.instance.as_mut().expect("Must have instance");
+        let runtime = self.runtime.as_mut().expect("Must have runtime");
```

Comment:
Better to return corresponding `Err`

Reply:
It, is not backend error, it's software engineer mistake .